core/tests/LowLevelTests/Op.hs

```diff
@@ -1,60 +1,81 @@
-lowLevelOpTests = testGroup "Synchronous unit tests of low-level Op interface"
-  [testCancelFromServer]
+lowLevelOpTests =
+  testGroup
+    "Synchronous unit tests of low-level Op interface"
+    [testCancelFromServer]
 
 testCancelFromServer :: TestTree
 testCancelFromServer =
   testCase "Client/Server - client receives server cancellation" $
-  runSerialTest $ \grpc ->
-    withClientServerUnaryCall grpc $
-    \(Client{..}, Server{..}, ClientCall{..}, sc@ServerCall{..}) -> do
-      serverCallCancel sc StatusPermissionDenied "TestStatus"
-      clientRes <- runOps unsafeCC clientCQ clientRecvOps
-      case clientRes of
-        Left x -> error $ "Client recv error: " ++ show x
-        Right [_,_,OpRecvStatusOnClientResult _ code _details] -> do
-          code @?= StatusPermissionDenied
-          return $ Right ()
-        wrong -> error $ "Unexpected op results: " ++ show wrong
-
+    runSerialTest $ \grpc ->
+      withClientServerUnaryCall grpc $
+        \(Client {..}, Server {..}, ClientCall {..}, sc@ServerCall {..}) -> do
+          serverCallCancel sc StatusPermissionDenied "TestStatus"
+          clientRes <- runOps unsafeCC clientCQ clientRecvOps
+          case clientRes of
+            Left x -> error $ "Client recv error: " ++ show x
+            Right [_, _, OpRecvStatusOnClientResult _ code _details] -> do
+              code @?= StatusPermissionDenied
+              return $ Right ()
+            wrong -> error $ "Unexpected op results: " ++ show wrong
 
 runSerialTest :: (GRPC -> IO (Either GRPCIOError ())) -> IO ()
 runSerialTest f =
-  withGRPC f >>= \case Left x -> error $ show x
-                       Right () -> return ()
+  withGRPC f >>= \case
+    Left x -> error $ show x
+    Right () -> return ()
 
-withClientServerUnaryCall :: GRPC
-                             -> ((Client, Server, ClientCall,
-                                  ServerCall ByteString)
-                                 -> IO (Either GRPCIOError a))
-                             -> IO (Either GRPCIOError a)
+withClientServerUnaryCall ::
+  GRPC ->
+  ( ( Client,
+      Server,
+      ClientCall,
+      ServerCall ByteString
+    ) ->
+    IO (Either GRPCIOError a)
+  ) ->
+  IO (Either GRPCIOError a)
 withClientServerUnaryCall grpc f = do
   withClient grpc clientConf $ \c -> do
     crm <- clientRegisterMethodNormal c "/foo"
-    withServer grpc serverConf $ \s ->
-      withClientCall c crm 10 $ \cc -> do
+    withServer grpc serverConf $ \s -> do
+      ccVar <- newEmptyMVar
+      bracket newEmptyMVar (\v -> putMVar v ()) $ \finished -> do
+        _ <- forkIO $
+          void $
+            withClientCall c crm 10 $ \cc -> do
+              putMVar ccVar cc
+              -- NOTE: We need to send client ops here or else `withServerCall` hangs,
+              -- because registered methods try to do recv ops immediately when
+              -- created. If later we want to send payloads or metadata, we'll need
+              -- to tweak this.
```

Comment:
This comment looks identical to the one below.  Do we need to keep both copies of the comment?

Reply:
definitely not, this also needs updating.